lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -307,29 +353,50 @@ static LogicalResult verifyComponentOp(ComponentOp op) {
                               "`clk`, `reset`, and output port `done`";
 }
 
+template <typename T>
```

Comment:
Please document this. 

Another option is using [SmallVectorImpl::append](https://llvm.org/doxygen/classllvm_1_1SmallVectorImpl.html#acd79f15ee5c01bfc3e2393233f40b764), which will also reserve space.